clients/src/test/java/org/apache/kafka/test/TestUtils.java

```diff
@@ -383,6 +383,10 @@ public static void waitForCondition(final TestCondition testCondition, final lon
         waitForCondition(testCondition, maxWaitMs, () -> conditionDetails);
     }
 
+    public static void waitForCondition(final TestCondition testCondition, String conditionDetails, final long maxWaitMs) throws InterruptedException {
```

Comment:
It's not recommend to keep 2 methods with same parameters but different order  for the simple reason just to reduce code change, especially in test module. so how about remove it?